### ISSUE TITLE
'class FEProblemBase' has no member named 'getUserObjectTempl'

### ISSUE THREAD

**Issue by zhouyukun93** [idaholab/moose#15339]
## Bug Description
1. error: 'class FEProblemBase' has no member named 'getUserObjectTempl'; did you mean 'getUserObjects'?
       return _fe_problem->getUserObjectTempl<T>(name);
2. error: expected primary-expression before '>' token
     return _fe_problem->getUserObjectTempl<T>(name);
## Steps to Reproduce
I was trying to compile MOOSE with SAM code by using the default Makefile under the SAM folder.  After I have input the command ‘’make -j 16”，the errors happened during the compilation.
## Impact
opt file can't be generated.

**Comment by dschwen** [comment]
Is everything up to date? There were fairly recent changes to these
functions.

**Comment by zhouyukun93** [comment]
Yes. The new MOOSE environment has been created on my Linux system using Conda. I am wondering maybe this new MOOSE version is not compatible with the SAM version I have? Or I have to roll back the MOOSE version?

**Comment by andrsd** [comment]
You should be using the MOOSE version that comes with SAM, i.e. from the submodule.  It is not guaranteed that the latest MOOSE works with SAM.  If you desire to use the current MOOSE version, you need to talk to SAM developers, so they update their code....
